src/Conversion/ONNXToKrnl/Tensor/Gather.cpp

```diff
@@ -142,24 +141,36 @@ struct ONNXGatherOpLowering : public OpConversionPattern<ONNXGatherOp> {
             // along axis of size s. It is an error if any of the index values
             // are out of bounds.
             // After the negative correction, the range should be [0, s-1]
-            Value upperBound = create.mem.dim(data, axisLit);
-            Value compareUpperBound =
-                create.math.slt(index.getValue(), upperBound);
             // Report onnx_node_name if the op has the attribute
-            std::string nodeNameStr = op->getName().getStringRef().str() + " ";
+            std::string nodeNameStr = "Warning: ";
+            nodeNameStr += op->getName().getStringRef().str() + " ";
             StringAttr nodeName =
                 op->getAttrOfType<mlir::StringAttr>("onnx_node_name");
             if (nodeName && !nodeName.getValue().empty()) {
-              nodeNameStr = nodeNameStr + nodeName.getValue().str();
+              nodeNameStr += nodeName.getValue().str();
             }
-            rewriter.create<cf::AssertOp>(loc, compareUpperBound,
-                nodeNameStr +
-                    " indices of GatherOp is larger than the upper bound");
+
+            Value upperBound = create.mem.dim(data, axisLit);
+            Value compareUpperBound =
+                create.math.sge(index.getValue(), upperBound);
             Value compareLowerBound =
-                create.math.sge(index.getValue(), zeroIE.getValue());
-            rewriter.create<cf::AssertOp>(loc, compareLowerBound,
-                nodeNameStr +
-                    " indices of GatherOp is less than the lower bound");
+                create.math.slt(index.getValue(), zeroIE.getValue());
+            Value outBound =
+                create.math.ori(compareUpperBound, compareLowerBound);
```

Comment:
please add `SCFBuilder` to create object, and then you can use 
```
  void ifThenElse(mlir::Value cond, SCFThenElseBodyFn thenFn,
      SCFThenElseBodyFn elseFn = nullptr) const;
```
to be in line with current practices.